P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/core/WindowGeneration.h

```diff
@@ -5,52 +5,81 @@
 #include "IO/Logger.h"
 
 namespace WindowGeneration {
-    [[nodiscard]] static std::vector<Series> GenerateWindows(const Series &series, uint length) {
+    /// <summary>
+    /// Generates a vector of sub-series out of a series, based on a specific window size.
+    /// </summary>
+    /// <param name="series">A vector of doubles</param>
+    /// <param name="length">An integer, representing the length of each window to be made</param>
+    /// <returns>A vector of sub-series</returns>
+    [[nodiscard]] static std::vector<Series> GenerateWindowsOfLength(const Series &series, uint length) {
         std::vector<Series> windows;
 
         if (series.empty() || length == 0)
             return windows;
 
-        for (uint i = 0; i + length <= series.size(); ++i) {
-            Series window;
-            const double yOffset = series[i]; // Always start a window at 0
-            for (uint offset = 0; offset < length; ++offset)
-                window.push_back(series[i + offset] - yOffset);
-            windows.push_back(window);
+        for (uint i = 0; i < series.size(); i++) {
+            if (i + length <= series.size()) {
+                Series window;
+                for (uint j = 0; j < length; j++)
+                    window.push_back(series[i + j]);
+                windows.push_back(window);
+            }
         }
 
         return windows;
     }
 
-    [[nodiscard]] static std::vector<Series> GenerateWindows(const Series &series, uint minLength, uint maxLength) {
+    /// <summary>
+    /// Generates a vector of sub-series out of a series, based on all possible window sized between a min and max length
+    /// </summary>
+    /// <param name="series">A vector of doubles</param>
+    /// <param name="minLength">An inclusive minimum length of a window</param>
+    /// <param name="maxLength">An inclusive maximum length of a window</param>
+    /// <returns>A vector of sub-series</returns>
+    [[nodiscard]] static std::vector<Series> GenerateWindowsOfMinMaxLength(const Series &series, uint minLength, uint maxLength) {
```

Comment:
Why change the naming scheme?

Reply:
I suppose? Although, I don't see how they do different things. They all generate a set of windows, based upon differing parameters. As such, they were named the same thing ... with different parameters.

The function I see breaking this pattern is the, now, solely named generate windows. Which also removes duplicates, behavior not seen in the other formerly named GenerateWindow functions.

Adding the parameter names to the functions seem rather redundant.

Reply:
One generates windows of a size n only.

One generates windows of all possible sizes between n and m (min and max length)

And one does the same as above, but also removes duplicates

I dont see a reason to not make the names more explicit.
However the last one should probably also be renamed, since it does all the duplicate removing thing.